src/middlewared/middlewared/plugins/chart_releases_linux/resources.py

```diff
@@ -27,6 +28,68 @@ async def pod_console_choices(self, release_name):
 
         return choices
 
+    @accepts(Str('release_name'))
+    async def pod_console_choices(self, release_name):
+        """
+        Returns choices for console access to a chart release.
+
+        Output is a dictionary with names of pods as keys and containing names of containers which the pod
+        comprises of.
+        """
+        return await self.retrieve_pod_with_containers(release_name)
+
+    @accepts(Str('release_name'))
+    async def pod_logs_choices(self, release_name):
+        """
+        Returns choices for accessing logs of any container in any pod in a chart release.
+        """
+        return await self.retrieve_pod_with_containers(release_name)
+
+    @private
+    async def validate_pod_log_args(self, release_name, pod_name, container_name):
+        choices = await self.pod_logs_choices(release_name)
+        if pod_name not in choices:
+            raise CallError(f'Unable to locate {pod_name!r} pod.', errno=errno.ENOENT)
+        elif container_name not in choices[pod_name]:
+            raise CallError(
+                f'Unable to locate {container_name!r} container in {pod_name!r} pod.', errno=errno.ENOENT
+            )
+
+    @accepts(
+        Str('release_name'),
+        Dict(
+            'options',
+            Str('pod_name', required=True, empty=False),
+            Str('container_name', required=True, empty=False),
+        )
+    )
+    @job(lock='dataset_export_keys', pipes=['output'])
+    def pod_logs(self, release_name, options):
+        """
+        Export logs of `options.container_name` container in `options.pod_name` pod in `release_name` chart release.
+
+        Please refer to websocket documentation for downloading the file.
+        """
+        self.middleware.call_sync(
+            'chart.release.validate_pod_log_args', release_name, options['pod_name'], options['container_name']
+        )
+
+        logs = self.middleware.call_sync(
+            'k8s.pod.get_logs', options['pod_name'], options['container_name'], get_namespace(release_name)
+        )
+        temp_path = None
+        try:
+            with tempfile.NamedTemporaryFile(mode='w', delete=False) as f:
+                temp_path = f.name
+                os.chmod(temp_path, 0o600)
+                f.write(logs)
+
+            with open(temp_path, 'rb') as f:
+                shutil.copyfileobj(f, job.pipes.output.w)
```

Comment:
You may just write directly to the pipe.

There is little point in piping `k8s.pod.get_logs` result since it is already loaded to the RAM entirely